CMakeLists.txt

```diff
@@ -18,7 +18,17 @@ set(custom_compiler_flags)
 include(CheckCCompilerFlag)
 option(ENABLE_CUSTOM_COMPILER_FLAGS "Enables custom compiler flags" ON)
 if (ENABLE_CUSTOM_COMPILER_FLAGS)
-    if (("${CMAKE_C_COMPILER_ID}" STREQUAL "Clang") OR ("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU"))
+    if ("${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
+        list(APPEND custom_compiler_flags
+            -std=c89
+            -pedantic
+            -Weverything
```

Comment:
`-Weverything` is not something I'm willing to put in there. I use it from time to time in order to find new warnings that I can use though.